### DISCUSSION TITLE
NNG (and nanomsg) Leadership & Continuity Planning

### DISCUSSION THREAD

**Discussion by gdamore** [nanomsg/nng#1782] [Announcements]
It's time.  We need to have a a continuity plan, in case something happens to me, and I'd like to have an advisory council as well, who can help answer questions and provide advice from different community view points.

First off, I'm not yielding up my BDFL status for NNG, nanomsg, or mangos.  I believe the project needs strong leadership, and I've seen other projects (including at least one well known project that I founded) flounder and flail when leadership decisions are relegated to consensus driven decision making.  (The problem with consensus based decisions is that it gives everyone a veto, and so no hard decisions can ever be made.  If you have some kind of majority rule then you wind up delegating decision making to folks who may not have any idea of the technical issues at hand, and a well-spoken but completely clueless person can drive a decision that is totally unworkable in actuality.)

If anyone doesn't like my leadership, they are of course free to fork.  That's the power of open source. 

Of course, forking is expensive, and its better if a continuity plan is in place to prevent the need to fork if something unfortunate happens that prevents me from being able to lead the project.

So let's start with an advisory board:

The advisory board duties shall include:

1. Advising the BDFL at various times and on various issues as the advisory board and the BDFL mutually find appropriate.
2. Upon the demise or permanent incapacity of the BDFL, shall either identify a new BDFL, or select a new form of governance.  Either option shall require approval of a 2/3 majority of the then serving advisory board members.
3. Upon the demise or permanent incapacity of the BDFL, until a new BDFL or alternative governance is formed, shall handle the day to day operation of the project (approving, and merging pull requests, etc.)
4. Meet periodically -- at least once per year -- with the BDFL and other advisory board members, for planning purposes for the coming year.  (Date to be specified -- meeting will be virtual.)
5.  Additional other duties as the BDFL and the advisory board mutually deem necessary.

Except for the noted 2/3 continuity votes, the advisory board decision making, when it occurs, shall be done by simple majority; however the BDFL retains the right to overrule any decision as an absolute authority over the project.

How to select members?

1. The BDFL has absolute authority to reject or approve for admission any individual for the advisory board.
2. The BDFL can add (subject to the approval of the prospective member) or remove a member at any time.
3. The advisory board shall consist only of real people (natural persons).
4. Anyone in the community may nominate any other person, including themselves.
5. We should aim for 5-7 members on the advisory board.
6. In the event that more candidates present than available positions on the board exist, the BDFL may run a community poll to select between some of the members.  Alternatively the BDFL may just select candidates using whatever criteria they deem fit.

So with that said, I'd like to open up to nominations.  Please post any nominations in this discussion.

**Comment by shikokuchuo** [comment]
Thanks @gdamore. I’d self-nominate, mainly as I’ve written the R language binding http://github.com/shikokuchuo/nanonext and I'd like to be able to represent the interests of my users.

Some of the binding work has been supported by the R Consortium, a project under the Linux Foundation. Amongst other things, it powers the ‘mirai’ async framework for parallel and distributed computing. This is used in high-performance computing contexts such as machine learning, Bayesian statistics and simulations pipelines in industries such as the life sciences etc.

If you’re open to diverse viewpoints, I’d be happy to provide one that’s probably a bit unique.

**Reply by gdamore** [reply]
Thank you @shikokuchuo  - stay tuned.  Selections will start once we have a few more parties here.  I know that there several people who have told me privately they would like to participate as well.

**Comment by neachdainn** [comment]
I would also like to self-nominate, for similar reasons. I maintain the Rust wrapper for NNG and I have an interest in making sure that NNG continues to be useful given the constraints of that language.

While C isn't my primary language, I understand it well enough that I should be capable of contributing to any discussions about the code itself. As far as networking goes, I'm in a similar spot as C: it's not my primary domain but I am close enough to it at `$DAYJOB` that I should be able to contribute something to discussions.

**Comment by JaylinYu** [comment]
Hi Garrett,  I feel cheerful that you finally making such a move and always being inspired by your wisdom and rich open-source experience. 
We, NanoMQ project, me, Jaylin from EMQ would like to lend our hands to help you keep NNG sustainable since it has become one of our pillars in edge computing. 
As per our discussions in the last two years, I believe we have established a good mutual understanding and acknowledgment. After our last sponsorship, we continually improving NanoMQ project, being a heavy user on NNG, both SP protocol and internal aio framework, we are more than happy to see it flourish and attract more developers from all over the world. We have achieved a great success in business world with NNG, would like to share this with community if possible.
C is our major language, we can also helps solve issues, develop QUIC/MQTT support, and other potential extensions as we discussed before.

**Comment by gdamore** [comment]
I've added @neachdainn to start with.  I'll reach out as I add more. :-)

**Comment by gdamore** [comment]
Welcome to @JaylinYu to the leadership team.  🎊
